src/AV/Video.cpp

```diff
@@ -1367,10 +1368,10 @@ int Video::addTimer(int delay)
 
 bool Video::addPacket(PacketQueue *queue, AVPacket *packet)
 {
-    av_dup_packet(packet);
+    AVPacket *dup = av_packet_clone(packet);
```

Comment:
So I am still a but puzzled by this one. 

The packet is retrieved with `av_read_frame(m_Container, &packet)` and the docs says : 

> If pkt->buf is NULL, then the packet is valid until the next av_read_frame() or until avformat_close_input(). Otherwise the packet is valid indefinitely. In both cases the packet must be freed with av_packet_unref when it is no longer needed.

So if `packet->buf` is NULL then the next call to av_read_frame() will invalidate it no mater if it's ref counted or not (well at least this is how I understand that)

@verpeteren Did a quick test, and it shows that packet->buf is never null (well at least in his test). So what I am suggesting is : 

- If packet->buf is NULL, nothing to do and no need to ref it again, the packet will be valid indefinitely
- If packet->buf is not NULL we need to duplicate it, because (as the docs says) the packet will no longer be valid on the next call to `av_read_frame()`